eta/core/video.py

```diff
@@ -4138,18 +4138,14 @@ def run(self, inpath, outpath, verbose=False):
             in_opts = self._in_opts
 
         # Automatically determine the starting number of the inpath for cases
-        # where it's a sequence pattern (e.g. %06d.jpg).  The default behavior
-        # from ffmpeg is to start at 0 and look in the range [0,4].  If the first
-        # matched pattern begins above 4 we want to explicitly set that.
-        if "-start_number" not in in_opts and is_supported_image_sequence(
-            inpath
+        # where it's a sequence pattern (e.g. %06d.jpg). The default behavior
+        # from ffmpeg is to start at 0 and look in the range [0, 4]. If the
+        # first matched pattern begins above 4 we want to explicitly set that.
+        if is_supported_image_sequence(inpath) and (
+            "-start_number" not in in_opts
         ):
             start_number = next(iter(etau.parse_pattern(inpath)), 0)
-
-            # Ensure in_opts modification is a per-call change
-            in_opts = in_opts.copy().extend(
-                ["-start_number", str(start_number)]
-            )
+            in_opts = in_opts.copy() + ["-start_number", str(start_number)]
```

Comment:
nit:

you don't need the `.copy()` here - `+` always creates a new list:
```
In [1]: mylist = ['foo', 'bar']

In [2]: newlist = mylist + ['baz', 'buzz']

In [3]: mylist
Out[3]: ['foo', 'bar']

In [4]: newlist
Out[4]: ['foo', 'bar', 'baz', 'buzz']
```

I would, however, suggest that you consider
`in_opts += ["-start_number", str(start_number)]` as it is significantly more performant.

unless you feel like you need a shallow copy of the list for some reason?

```
In [5]: mylist += ['baz', 'buzz']

In [6]: mylist
Out[6]: ['foo', 'bar', 'baz', 'buzz']
```

for example purposes:
```
In [7]: otherlist = ['foo', 'bar']

In [8]: otherlist = otherlist.copy() + ['baz', 'buzz']

In [9]: otherlist
Out[9]: ['foo', 'bar', 'baz', 'buzz']
```

though you could continue to use `extend` if we got rid of the copy...  that just feels like an odd syntax (very nonpythonic):

```
In [10]: extendlist = ['foo', 'bar']

In [11]: extendlist.extend(['baz', 'buzz'])

In [12]: extendlist
Out[12]: ['foo', 'bar', 'baz', 'buzz']
```

Reply:
Yes, I may be bikeshedding list expansion.

Reply:
`FFmpeg` is a bit weird: `in_opts` is shared between calls to `run()` so we need to avoid editing in-place so that each call to `run()` is independent.

Happy to (a) remove `copy()` though and (b) give you a hard time for worrying about list creation right before a subprocess is opened 😉